tests/test_facebook_pagination.py

```diff
@@ -0,0 +1,68 @@
+import test_client as tc
+import time
+import unittest
+
+from datetime import datetime as dt
+
+from tap_tester.base_suite_tests.pagination_test import PaginationTest
+from tap_tester import connections, runner, menagerie, LOGGER
+
+from base_new_frmwrk import FacebookBaseTest
+
+fb_client = tc.TestClient()
+
+
+class FacebookDiscoveryTest(PaginationTest, FacebookBaseTest):
+    """Standard Pagination Test"""
+
+    @staticmethod
+    def name():
+        return "tt_facebook_pagination"
+    def streams_to_test(self):
+        # TODO ads_insights empty for account, no post via API, spike on generating data
+        return {'adcreative', 'ads', 'adsets', 'campaigns'}
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """
+        Setup for tests in this module.
+        """
+        if PaginationTest.synced_records and PaginationTest.record_count_by_stream:
+            return
+
+        # instantiate connection
+        conn_id = connections.ensure_connection(self)
+
+        # run check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # table and field selection
+        test_catalogs = [catalog for catalog in found_catalogs
+                         if catalog.get('stream_name') in self.streams_to_test()]
+
+        # non_selected_fields are none
+        self.perform_and_verify_table_and_field_selection(conn_id, test_catalogs)
+
+        # ensure there is enough data to paginate
+        start_date_dt = self.parse_date(self.get_properties()['start_date'])
```

Comment:
In the new framework I though we usually define a `self.start_date` in base and then can access that directly instead of through `get_properties`.  That way you can also customize the `start_date` directly in a test easier.

Reply:
Updated start_date to follow new pattern